containers/podman-desktop
# fix: Incorrect pod status

### What does this PR do?

ContainerGroupIcon was just a PodIcon, setting the status to running or stopped based on whether all containers in the group were running or not. Pods have a degraded state where only some of the containers are running.

Instead of trying to make this icon more complicated, this change falls back to directly using the PodIcon and the status of the container group. There are three kinds of groups, with the status now set in container-utils.ts:
- standalone containers - set to the container status
- pods - set to pod status
- compose groups - wait until the group is created, then use the same logic as before

### Screenshot/screencast of this PR

<img width="248" alt="Screenshot 2023-05-08 at 8 21 50 AM" src="https://user-images.githubusercontent.com/19958075/236822550-43cddf84-090d-4c05-a03b-406b1729038c.png">

### What issues does this PR fix or reference?

Fixes #2340.

### How to test this PR?

Start a degraded pod (e.g. stop a container, or create a pod where one of the containers won't start) and check the status in the Containers page.
